components/apimgt/org.wso2.carbon.apimgt.micro.gateway/org.wso2.carbon.apimgt.micro.gateway.usage.publisher/src/test/java/org/wso2/carbon/apimgt/micro/gateway/usage/publisher/APIMgtUsageFileDataPublisherTest.java

```diff
@@ -163,4 +167,15 @@ public void publishEvent_withAlertTypeDTO() throws Exception {
         apiMgtUsageFileDataPublisher.init();
         apiMgtUsageFileDataPublisher.publishEvent(alertTypeDTO);
     }
+
+    @AfterClass
+    public static void cleanUp() throws Exception {
+        //Cleaning the file
+        String carbonHome = System.getProperty(Constants.CARBON_HOME);
+        PrintWriter writer = new PrintWriter(
+                carbonHome + File.separator + "api-usage-data" + File.separator + "api-usage-data.dat");
+        Thread.sleep(3000);
```

Comment:
Shall we explain why we have a Thread.sleep here?

Reply:
@npamudika I think we don't need this

Reply:
Removed the Thread.sleep().